better-code/src/chapter-2-contracts.md

```diff
@@ -0,0 +1,1129 @@
+  @requires(i >= 0 && i < self.count)
+  fun getNth(i: Integer): T
+
+  ...
+}
+```
+
+The idea started with Bertrand Meyer's own Eiffel language, and was
+picked up by many others, including D, Scala, Kotlin, and
+Clojure. Other languages, like Rust and Python, have contract
+libraries that provide a near-native contract programming experience.
+
+If you use one of these languages, fantastic; *absolutely do* leverage
+those features and libraries.  That can reduce the amount of pure
+documentation you need to write and add automated contract checking at
+runtime. But there are caveats:
+
+1. **Contracts are fundamentally documentation**, even if they're
+   expressed in code, so they must appear in the API descriptions
+   consumed by client programmers. If you're using contract checking
+   with automated documentation extraction tools make sure they expose
+   the contract code along with the API.  If not, you need to repeat
+   the complete contract in documentation.
+
+2. Some contracts are better expressed in English than as code,
+
+3. Some contracts are impossible to express as code.  We'll
+   see an example in a moment.
+
+We'll discuss contract checking as a separate topic in the next
+chapter.
+
+#### Document Everything
+
+So **every declaration needs to be documented**. We realize that's not
+standard industry practice, and to many people it sounds like a huge
+burden!  Read on, though, and we'll show you how it can be done both
+thoroughly *and economically*.  To see that it's possible, look again
+at the documentation shown for `PairArray`.  It tells you everything
+you need to know to use or test each declared element, but is not
+burdensome to write.
+
+Comprehensively documenting your declarations is not only practical
+and necessary, but it has *benefits* beyond correctness and local
+reasoning: it helps you improve your APIs and can even increase overall
+development speed. But you'll only see those benefits if you document *as
+you code*. If documentation is an afterthought, it can only ever be
+experienced as overhead.
+
+If you take this discipline seriously, you'll begin to notice that the
+poorly designed or named components haven't been documented clearly
+and concisely.  Sometimes that documentation is obviously unachievable
+because the design simply has too many wrinkles.  In other cases
+you'll find the design can be salvaged by driving it towards a simpler
+contract. Improvements in development speed come from arriving at the
+right design earlier, since a poor experience for readers of your
+documentation becomes a “code smell.”
+
+#### Use Documentation Comments
+
+Contract documentation should always go in your code as comments,
+because:
+
+1. That puts the two things that should correspond—documentation and
+   implementation—next to one another, so you can see when they don't
+   match up. People sometimes complain that docs go out of date, but
+   that's kinda the point: without the ability to see that
```

Comment:
```suggestion
   that's exactly the point: without the ability to see that
```

I agonized over this, because I like the original vibe, but ultimately I think the slang is more distracting than useful.